webui/src/js/utils/k8s-domain-resource-generator.js

```diff
@@ -247,7 +247,22 @@ define(['models/wkt-project', 'utils/k8s-domain-configmap-generator', 'js-yaml',
     }
 
     function getServerPodForCluster(cluster) {
-      return _getServerPod(getJavaOptionsForCluster(cluster), getUserMemArgsForCluster(cluster), getKubernetesResourcesForCluster(cluster));
+      const serverPod = {};
+      const env = [];
+      addIfNotNull(env, 'JAVA_OPTIONS', getJavaOptionsForCluster(cluster));
+      addIfNotNull(env, 'USER_MEM_ARGS', getUserMemArgsForCluster(cluster));
+      if (env.length) {
+        serverPod.env = env;
+      }
+
+      const resources = getKubernetesResourcesForCluster(cluster);
+      if (resources) {
+        serverPod.resources = resources;
+      }
+
+      serverPod.affinity = _getAffinityForServerPod(100);
+
+      return Object.keys(serverPod).length > 0 ? serverPod : null;
```

Comment:
Why duplicate all the code from _getServerPod() just to add one line?  Why not just do this:

```
function getServerPodForCluster(cluster) {
    serverPod = _getServerPod(getJavaOptionsForCluster(cluster), getUserMemArgsForCluster(cluster), getKubernetesResourcesForCluster(cluster)) || {};
    serverPod.affinity = _getAffinityForServerPod(100);
    return serverPod;
}
```

Reply:
Server Pod might be empty, and I did not know how to add const serverPod = {}

Reply:
resolved code per suggestions